reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscriber.scala

```diff
@@ -126,9 +134,10 @@ object StreamSubscriber {
 
     sealed trait State
     case object Uninitialized extends State
-    case class Idle(sub: Subscription) extends State
+    case class Idle(sub: Subscription, buffer: Vector[A]) extends State
```

Comment:
Change `Vector[A]` to `Chunk[A]` and use `++` to build up the buffer.

Reply:
https://github.com/typelevel/fs2/pull/2666/commits/769b2fe568a2627b454fbbc7b3f6bcfb2c17968a